### PR TITLE
18 feature svg nodes for center

### LINKED ISSUE #18: [Feature] Node and Edge drawing capabilities
### Description

Setting up the SVG drawing board and implementing drawing capabilities of states (locations) and edges (actions).

### Reason For Request
It is a required feature to be able to create and change components and systems.

### PR DESCRIPTION
**Pull request by TSKsmiley**
Closes #18 

### Changes/Additions:

- made nodes and edges and nails 
- not the ability to draw them (end user) only for rendering them
- pan and zoom

### CODE REVIEW

File: src/lib/components/svg-view/Label.svelte [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+		y={position.y - 10}
+		width={text.length * 10}
+		height="20"
+		fill="darkgray"
+		text-anchor="middle"
+		stroke-width="2"
+	/>
+
+	<!-- text label -->
+	<text x={position.x} y={position.y + 5} text-anchor="middle" role="none">
```
Comment (atjn) [inline src/lib/components/svg-view/Label.svelte:35]: What is this random `+ 5`? If it is necessary, please define it as a variable with an explanation.

File: src/lib/components/svg-view/Node.svelte [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+<DraggableSVG bind:position>
+	<circle
+		cx={position.x}
+		cy={position.y}
+		r={radius}
+		fill="gray"
+		role="none"
+	/>
+	<text x={position.x} y={position.y + 5} text-anchor="middle" role="none"
```
Comment (atjn) [inline src/lib/components/svg-view/Node.svelte:18]: Same comment, fix random `+ 5`

File: src/lib/globalState/activeModel.ts [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+import type { iLocation } from "$lib/interfaces/iLocation";
+import type { iEdge } from "$lib/interfaces/iEdge";
+import {
+	LocationType,
+	PropertyType,
+	Status,
+	Urgency,
+} from "$lib/classes/automaton";
+
+// TODO: Move this class to its own file
```
Comment (atjn) [inline src/lib/globalState/activeModel.ts:11]: Remove TODO that is now TODID

File: src/routes/+page.svelte [hunk 4]
```diff
@@ -58,7 +57,7 @@
 </script>
 
 <!-- Top navigation Panel -->
-<nav id="main-nav"></nav>
+<nav id="main-nav">Main Nav</nav>
```
Comment (atjn) [inline src/routes/+page.svelte]: Remove this to not have merge conflicts with Console

**Review by atjn — CHANGES_REQUESTED** [review]

File: src/lib/components/svg-view/Label.svelte (hunk 1 above)
Comment (TSKsmiley) [reply src/lib/components/svg-view/Label.svelte:35]: it moves the text into the rectangle, how would you suggest going about the variable just in the script?

Comment (atjn) [reply src/lib/components/svg-view/Label.svelte:35]: Yes put a const in the script

File: src/lib/globalState/scaleStore.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import { writable, type Writable } from "svelte/store";
+
+/**
+ * Dines the current scaling of the main SVG view.
```
Comment (TSKsmiley) [inline src/lib/globalState/scaleStore.ts:4]: what does it mean to Dine is it like serving but posh?

Comment (atjn) [reply src/lib/globalState/scaleStore.ts:4]: Whoops. Defines 😂

File: src/lib/components/svg-view/DraggableSVG.svelte [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+	let controller: AbortController;
+	let signal: AbortSignal;
+
+	//Sets up eventlisteners when the mouse is pressed down on the svg
+	function onPointerDown() {
+		controller = new AbortController();
+		signal = controller.signal;
+		window.addEventListener("pointermove", onPointerMove, { signal });
+		window.addEventListener("pointerup", onPointerUp, { signal });
```
Comment (atjn) [inline src/lib/components/svg-view/DraggableSVG.svelte:16]: Pointervents cannot be treated exactly the same as Mouseevents. For example, I don't see an event listener for `pointercancel`. Did you read and understand the example I pointed you to? https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web/blob/main/src/routes/%2Bpage.svelte

File: src/lib/components/svg-view/Label.svelte [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+	<!-- line from parrent to text label, make dotted-->
+	<line
+		stroke-dasharray="5,5"
+		d="M5 20 l215 0"
+		x1={parentPosition.x}
+		y1={parentPosition.y}
+		x2={position.x}
+		y2={position.y}
+		stroke="gray"
+		stroke-width="2"
```
Comment (atjn) [inline src/lib/components/svg-view/Label.svelte:22]: As I mentioned in the original review, you need to move these variables to CSS in many of your files, not just at the exact point I have written this comment.

**Review by atjn — CHANGES_REQUESTED** [review]

File: src/lib/components/svg-view/DraggableSVG.svelte (hunk 6 above)
Comment (TSKsmiley) [reply src/lib/components/svg-view/DraggableSVG.svelte:16]: @AlbertHald dette sortere under dig

Comment (AlbertHald) [reply src/lib/components/svg-view/DraggableSVG.svelte:16]: Personligt nej tak, men også ja tak @atjn

Comment (TSKsmiley) [reply src/lib/components/svg-view/DraggableSVG.svelte:16]: is this solved now? @atjn

**Review by atjn — APPROVED** [review]
image

**Comment by TSKsmiley** [comment]
oh btw @ECDAR-AAU-SW-P5/group5 så laver vi connection til det der dataflow noget i en ny pr

**Review by DenFlyvendeGed — APPROVED** [review]
LGTM

**Pull request merged**